src/services/smart-contracts/actions/erc20.js

```diff
@@ -0,0 +1,32 @@
+
+export const isEthAddressERC20 = async (addr = '') => {
+	try {
+		if (isEthAddress(addr)) {
+			const eth = await getEthers()
+			const contract = new ethers.Contract(addr, ERC20TokenABI, eth.provider)
+			await contract.totalSupply()
+			await contract.balanceOf(constants.AddressZero)
+			await contract.allowance(constants.AddressZero, constants.AddressZero)
+			return true
+		}
+	} catch (error) {
+		return false
+	}
+}
+
+export const isConnectionLost = async (addr = '') => {
```

Comment:
Can you do this function common not only for ERC20 tokens and use `provider.getBalance(addr)` if there is no better way. And what if `isEthAddress` returns `false`

Reply:
Where should I put the function? Should I move this to `helpers/validators.js` or should I create an `actions/common.js`?

Reply:
smart-contracts/action/common